### PR TITLE
feat: (storage) remove validation of max 2 items for data_locations in custom_placement_config field

### PR DESCRIPTION
**Pull request by gurusai-voleti**
**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:bug
storage: lowered the minimum required items for `custom_placement_config.data_locations` from 2 to 1, and removed the Terraform-enforced maximum item limit for the field in `google_storage_bucket`
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl [hunk 1]
```diff
@@ -710,6 +710,24 @@ func TestAccStorageBucket_storageClass(t *testing.T) {
 	})
 }
 
+func TestAccStorageBucket_invalidStorageClass(t *testing.T) {
```
Comment (kautikdk) [inline mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl:713]: What exactly is this test validating and how is it related to the change in this PR?

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl [hunk 2]
```diff
@@ -710,6 +710,24 @@ func TestAccStorageBucket_storageClass(t *testing.T) {
...
+	bucketName := fmt.Sprintf("tf-test-acc-bucket-%d", acctest.RandInt(t))
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccStorageBucketDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccStorageBucket_storageClass(bucketName, "HOTLINE", "US"),
+				ExpectError: regexp.MustCompile("The storage class you specified is not valid"),
```
Comment (kautikdk) [inline mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl:725]: Why are we putting error validation in the acceptance test when `storage_class` attribute doesn't have any special behavior? i.e. DiffSupressFunc,StateFunc,etc.

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl (hunk 1 above)
Comment (gurusai-voleti) [reply mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl:713]: the change is to allow atleast one location in placement-config to allow creation of RAPID buckets, this test case is  validating the API response if any invalid storage classes are sent

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl (hunk 2 above)
Comment (NickElliot) [reply mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl:725]: We don't need a test to validate API error messages in Terraform, negative test cases are used for when the validation or other special behavior is contained within Terraform as Kautik mentioned

**Review by NickElliot — CHANGES_REQUESTED** [review]
that test case is unneeded, removing the max items limit should be fine as is as long as the api does accept it

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl (hunk 2 above)
Comment (gurusai-voleti) [reply mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.tmpl]: removed the test case

**Review by kautikdk — APPROVED** [review]
LGTM for the change!
I would leave the validation constraints on your decision. Not sure if there should be max limit or not as I'm not aware of the requirement.

**Comment by gurusai-voleti** [comment]
> LGTM for the change! I would leave the validation constraints on your decision. Not sure if there should be max limit or not as I'm not aware of the requirement.

no need of max limit as its validated by backend API

**Review by NickElliot — APPROVED** [review]
all lgtm! sorry for last delay, was flagging an issue with the breaking change detector to our team
